### PR TITLE
Add drop-check for index creation in cluster

### PR DESCRIPTION
**Pull request by danielhlarkin**
### Scope & Purpose

Add check in coordinator index creation process to verify that the index has not been dropped between polling attempts. Previously, this could result in the creation request hanging and timing out instead of recognizing that a simultaneous drop request had been made.

- [x] Bug-Fix for *devel-branch* (i.e. no need for backports?)
- [x] The behavior in this PR can be (and was) *manually tested* (support / qa / customers can test it)

#### Related Information

- [x] There is an internal planning ticket: https://github.com/arangodb/release-3.5/issues/101

### Testing & Verification

http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/4780/?auto_refresh=false
(Windows build is red due to Jenkins issues.)

### Documentation

- [x] Added a *Changelog Entry*

### CODE REVIEW

File: arangod/Cluster/ClusterInfo.cpp [hunk 1]
```diff
@@ -2878,7 +2879,41 @@ Result ClusterInfo::ensureIndexCoordinatorInner(  // create index
...
+        if (result.successful()) {
+          auto indexes = result.slice()[0].get(
+              std::vector<std::string>{AgencyCommManager::path(), "Plan",
+                                       "Collections", databaseName,
+                                       collectionID, "indexes"});
+
+          bool found = false;
+          if (indexes.isArray()) {
+            for (size_t i = 0; i < indexes.length(); i++) {
+              VPackSlice v = indexes.at(i);
```
Comment (graetzer) [inline arangod/Cluster/ClusterInfo.cpp]: VPackArrayIterator

File: arangod/Cluster/ClusterInfo.cpp [hunk 2]
```diff
@@ -2878,7 +2879,41 @@ Result ClusterInfo::ensureIndexCoordinatorInner(  // create index
...
+              VPackSlice const k = v.get(StaticStrings::IndexId);
+              if (k.isString() && idString == k.copyString()) {
```
Comment (graetzer) [inline arangod/Cluster/ClusterInfo.cpp]:
```suggestion
              if (k.isString() && k.isEqualString(idString)) {
```

File: arangod/Cluster/ClusterInfo.cpp (hunk 1 above)
Comment (danielhlarkin) [reply arangod/Cluster/ClusterInfo.cpp]: For some reason using the `VPackArrayIterator` and a `continue` to skip was terminating the loop early. This way worked reliably. Does it make much difference functionally?

Comment (graetzer) [reply arangod/Cluster/ClusterInfo.cpp]: array iterator is faster ?

Comment (danielhlarkin) [reply arangod/Cluster/ClusterInfo.cpp]: Okay, let me try again and see if it works reliably now. (Might have been something else confounding my investigation.)

**Comment by danielhlarkin** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/4804/
